openerp/addons/base/ir/ir_actions.py

```diff
@@ -651,6 +652,26 @@ def _get_states_wrapper(self, cr, uid, context=None):
         'use_write': 'current',
     }
 
+    @openerp.api.constrains('code')
+    def _check_python_code(self):
+        for action in self.filtered('code'):
+            try:
+                ast.parse(action.code)
```

Comment:
Why use ast.parse instead of test_expr which validates not only that it's python code but that it's python code we allow executing?